packages/nibijs/src/query/query.ts

```diff
@@ -102,4 +105,38 @@ export class NibiruQueryClient extends StargateClient {
       })
     }
   }
+
+  /** getTxByHash: Query a transaction (tx) using its hexadecial encoded tx hash.
+   * A tx hash uniquely identifies a tx on the blockchain.
+   *
+   * The hex-encoded tx hash is:
+   * - An unambiguous representation of the SHA-256 cryptographic hash in the
+   *   consensus layer.
+   * - Well-suited for human-facing applications, as it is easier to work with
+   *   than bytes.
+   *
+   * @example
+   * const txHash = "7A919F2CC9A51B139444F7D8E84A46EEF307E839C6CA914C1A1C594FEF5C1562"
+   * const txRespResult = await getTxByHash(txHash)
+   * */
+  getTxByHash = (txHashHex: string): Promise<Result<TxResponse>> =>
+    Result.ofSafeExecAsync(async () => {
+      const resBz = hexToBytes(txHashHex)
+      if (resBz.ok) {
+        return await this.tm.tx({ hash: resBz.ok })
+      }
+      throw resBz.err
```

Comment:
Throw statement should be modified to not throw

Reply:
Since the `throw` is inside a `Result.of*`, it's already inside a try-catch, so it won't actually throw.

Reply:
Can we move the throw to be inside the try/catch based on the result of this function?

That way the throw is ensured to be called within a try/catch

Reply:
I'm not following what u mean

Reply:
It's guaranteed to be in a try-catch because it's passed for lazy execution inside the `Result.of*` call